osbuild/util/ostree.py

```diff
@@ -215,15 +216,27 @@ def parse_input_commits(commits):
 def deployment_path(root: PathLike, osname: str, ref: str, serial: int):
     """Return the path to a deployment given the parameters"""
 
-    base = os.path.join(root, "ostree")
-
-    repo = os.path.join(base, "repo")
-    stateroot = os.path.join(base, "deploy", osname)
-
-    commit = rev_parse(repo, ref)
-    sysroot = f"{stateroot}/deploy/{commit}.{serial}"
-
-    return sysroot
+    if osname == "" and ref == "" and serial == None:
+        # there are many files that end with '.0' so narrow
+        # the search by providing the commit of the deployment
+        commit = rev_parse(root + "/ostree/repo", "ostree/1/1/0")
+        filenames =  glob(root + f'/**/{commit}.0', recursive=True)
```

Comment:
I feel like it should be simpler than this because we can just use look in the deploy directory:

```
>>> glob.glob(f'/ostree/deploy/*/deploy/*.0', recursive=True)
['/ostree/deploy/fedora-coreos/deploy/164a9a34590640f49fd9d8258f236d0b5097872464f11fb968dc40e04e1a51d4.0', '/ostree/deploy/fedora-coreos/deploy/373c4386e89adb0396039419f9b625ada8c7af9fae18b39822f7f1251d8c0894.0']
```

This example returns two entries, but that's just because I'm operating on system that has been up for a while and has been updated.